test_framework/terraform/aws/sles/k3s_instaces.tf

```diff
@@ -98,8 +98,13 @@ resource "null_resource" "rsync_kubeconfig_file" {
   ]
 
   provisioner "remote-exec" {
-    inline = ["until([ -f /etc/rancher/k3s/k3s.yaml ] && [ `sudo /usr/local/bin/kubectl get node -o jsonpath='{.items[*].status.conditions}'  | jq '.[] | select(.type  == \"Ready\").status' | grep -ci true` -eq $((${var.lh_aws_instance_count_controlplane} + ${var.lh_aws_instance_count_worker})) ]); do echo \"waiting for k3s cluster nodes to be running\"; sleep 2; done"]
 
+    inline = [
+      "while [ ! -f /tmp/signal ]; do echo \"waiting for packages installed\"; sleep 2; done",
```

Comment:
Suggest seeing if it's possible to use `cloud-init status` to check the provisioning status. 

ref: 
- https://cloudinit.readthedocs.io/en/latest/topics/cli.html#status
- (error code will be returned if error) https://github.com/delphix/cloud-init/blob/master/cloudinit/cmd/status.py#L178